### ISSUE TITLE
Inconsistent naming of model contexts cause issues with table prefixes.

### ISSUE THREAD

**Issue by wjbuys** [stffn/declarative_authorization#26]
When recursing into obligations to check authorization, an incorrect context name is generated. This is caused by using ActiveRecord::Base#table_name for the context name, instead of the model name tableized as is used everywhere else. The following patch should fix that:

```
--- /lib/declarative_authorization/authorization.rb
+++ /lib/declarative_authorization/authorization.rb
@@ -614,7 +614,7 @@
         @context ||= begin
           rule_model = attr_validator.context.to_s.classify.constantize
           context_reflection = self.class.reflection_for_path(rule_model, path + [hash_or_attr])
-          context_reflection.klass.table_name.to_sym
+          context_reflection.klass.to_s.tableize.to_sym
         rescue # missing model, reflections
           hash_or_attr.to_s.pluralize.to_sym
         end
```

**Comment by stffn** [comment]
More consistent deriving of context from model class

Closed by c3a2eae5ea3e87f7c58af9fe7398a9a512e61f18 [wjbuys]

**Comment by wjbuys** [comment]
Hey, any chance you could release an updated gem that fixes this? Thanks!
